File: src/prod/src/ServiceModel/reliability/failover/ScalingTriggerKind.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"
#include "ScalingTriggerKind.h"

using namespace std;
using namespace Common;
using namespace Reliability;

void ScalingTriggerKind::WriteToTextWriter(__in TextWriter & w, Enum const & val)
{
    switch (val)
    {
    case ScalingTriggerKind::Invalid:
        w << L"Invalid";
        return;
    case ScalingTriggerKind::AveragePartitionLoad:
        w << L"AveragePartitionLoad";
        return;
    case ScalingTriggerKind::AverageServiceLoad:
        w << L"AverageServiceLoad";
        return;
    }
}

File: src/prod/src/managed/Api/src/System/Fabric/Description/ScalingTriggerKind.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Description
{
    using System.Fabric.Interop;

    /// <summary>
    /// <para>Enumerates the available triggers for scaling.</para>
    /// </summary>
    public enum ScalingTriggerKind
    {
        /// <summary>
        /// <para>All Service Fabric enumerations have a reserved Invalid flag.</para>
        /// </summary>
        Invalid = NativeTypes.FABRIC_SCALING_TRIGGER_KIND.FABRIC_SCALING_TRIGGER_KIND_INVALID,

        /// <summary>
        /// <para>Indicates a trigger where scaling decisions are made based on average load of a partition.</para>
        /// </summary>
        AveragePartitionLoadTrigger = NativeTypes.FABRIC_SCALING_TRIGGER_KIND.FABRIC_SCALING_TRIGGER_KIND_AVERAGE_PARTITION_LOAD,

        /// <summary>
        /// <para>Indicates a trigger where scaling decisions are made based on average load of a service.</para>
        /// </summary>
        AverageServiceLoadTrigger = NativeTypes.FABRIC_SCALING_TRIGGER_KIND.FABRIC_SCALING_TRIGGER_KIND_AVERAGE_SERVICE_LOAD
    }

}